### PR TITLE
fix `EventSource requests are broken under proxy` (close #1106)

### LINKED ISSUE #1106: EventSource requests are broken under proxy
Based on the https://github.com/DevExpress/testcafe/issues/1458.

1) Open the Network tab in the browser devtools.
2) Open the https://testcafe-spike.qantasassure.com/health-insurance/quote page under playground
3) Fill all fields and press the `Continue` button
4) Press the `Continue As A Guest button`
5) See in the Network tab that requests are broken. It seems something went wrong during server-side processing

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

File: src/client/sandbox/node/window.js [hunk 1]
```diff
@@ -166,8 +166,12 @@ export default class WindowSandbox extends SandboxBase {
         }
 
         if (window.EventSource) {
-            window.EventSource           = url => new nativeMethods.EventSource(getProxyUrl(url));
-            window.EventSource.prototype = nativeMethods.EventSource.prototype;
+            window.EventSource            = url => new nativeMethods.EventSource(
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: `window.EventSource` has two arguments in constructor - https://html.spec.whatwg.org/multipage/comms.html#the-eventsource-interface

File: test/client/fixtures/sandbox/node/classes-test.js [hunk 2]
```diff
@@ -56,6 +56,26 @@ if (window.EventSource) {
...
         ok(eventSource instanceof EventSource);
     });
+
+    test('should have static constants (GH-1106)', function () {
+        ok(EventSource.hasOwnProperty('CONNECTING'));
+        ok(EventSource.hasOwnProperty('OPEN'));
+        ok(EventSource.hasOwnProperty('CLOSED'));
+    });
+
+    test('should process url to proxy with special flag (GH-1106)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/classes-test.js]: Add tests for the cases when EventSource is closed:
-  from client - `eventSource.close()`
- from server - status code does not equal `200`

**Comment by LavrovArtem** [comment]
FPR

File: test/client/fixtures/sandbox/node/classes-test.js [hunk 3]
```diff
@@ -56,6 +56,59 @@ if (window.EventSource) {
...
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/classes-test.js]: Add [CORS handling](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/request-pipeline/index.js#L50) and replace proxied url for `origin` property for EventSource requests.
Example:
```js
var eventSource= new EventSource("http://cross-domain.com", {
  withCredentials: true
});
eventSource.addEventListener('message', function(e) {
  if (e.origin === 'http://origin-site-url') ...;
});
```

Also let's add more real life tests:
- `EventSource` for same domain request.
- `EventSource` for cross domain request. 
- checking `origin` property

Comment (LavrovArtem) [inline test/client/fixtures/sandbox/node/classes-test.js]: It sounds like an unrelated problem. May be we need to create a separate issue.

Comment (LavrovArtem) [inline test/client/fixtures/sandbox/node/classes-test.js]: What do you think?

Comment (miherlosev) [inline test/client/fixtures/sandbox/node/classes-test.js]: Ok. I've created the separate issue - https://github.com/DevExpress/testcafe-hammerhead/issues/1114

**Pull request merged**